### PR TITLE
SPOT-230 Open Data Model (ODM) Setup

### PR DESCRIPTION
**Pull request by TaddWood**
This PR contains the  necessary scripts and statements to construct the Open Data Model (ODM) as specified in the ODM document. There are currently 6 tables that are provided in the ODM: event, user_context, endpoint_context, network_context, threat_intelligence_context, and vulnerability context. The details and schemas for each of these can be found in the ODM document. Both Parquet and Avro storage options are available when building the tables.  The tables are built using the odm_setup.sh script found in this PR, which requires a format argument to be passed in (i.e. ./odm_setup.sh pqt OR ./odm_setup.sh avro).  A README is also included as a reference guide for anyone look to setup the ODM tables.

### CODE REVIEW

File: spot-setup/odm/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+spot-setup/odm is licensed under Apache Version 2.0
+
+## Contributing
+
+Create a pull request and contact the maintainers.
+
+## Issues
+
+Report issues at the Apache Spot [issues] (https://github.com/Open-Network-Insight/open-network-insight/issues) page.
```
Comment (natedogs911) [inline spot-setup/odm/README.md]: @TaddWood i'm risking being redundant here but this should be changed as well, 
https://issues.apache.org/jira/projects/SPOT/issues

Comment (TaddWood) [reply spot-setup/odm/README.md]: @natedogs911 no worries, this one has been updated as well.  Thanks for the heads up.

File: spot-setup/odm/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+The main script in the repository is **odm_setup.sh** which is responsible of loading environment variables,
+creating folders in Hadoop for the ODM data (event, user_context, endpoint_context, network_context, threat_intelligence_context, vulnerability_context),
+create the Hive database,
+and finally execute hive query scripts that creates Hive tables needed to access Apache Spot data through the ODM.
+
+## Environment Variables
+
+**spot.conf** is the file storing the variables needed during the installation process including node assignment, User interface, Machine Learning and Ingest gateway nodes.
+This file also contains sources desired to be installed as part of Apache Spot, general paths for HDFS folders, Kerberos information and local paths in the Linux filesystem for the user as well as for machine learning, ipython, lda and ingest processes.
+To read more about these variables, please review the [wiki] (https://github.com/Open-Network-Insight/open-network-insight/wiki/Edit%20Solution%20Configuration).
```
Comment (natedogs911) [inline spot-setup/odm/README.md]: The ONI wiki, which may have some valuable information that should be pulled into the Spot docs or perhaps website, Is definitely out of date.

This is probably the best replacement for this link
http://spot.incubator.apache.org/doc/#configuration

**Comment by natedogs911** [comment]
The ASF header needs to be applied to all but `odd_setup.sh`, I'm not sure if it's acceptable to do this in a later PR. Perhaps some feedback from mentors/commiters would help there.

here's an example:
https://github.com/apache/incubator-spot/blob/2ebe57238614b6aa1162c6dc30758c516ae79273/spot-setup/create_dns_parquet.hql#L2

File: spot-setup/odm/README.md (hunk 2 above)
Comment (TaddWood) [reply spot-setup/odm/README.md]: @natedogs911 no worries. The link has been updated.

**Comment by TaddWood** [comment]
@natedogs911 I can make that modification.  Do we need to add the ASF header to the avro schemas as well?

**Comment by jnatkins** [comment]
@TaddWood I don't think that's possible, as JSON files do not support comments. My understanding is that avsc files would need to just be excluded from the RAT plugin that most projects use to enforce license compliance.

**Comment by TaddWood** [comment]
@jnatkins thanks for the clarification.  I'll get the ASF headers on the .sql files now and leave the avro schemas as is.

**Comment by mhicks-cloudera-com** [comment]
+1

**Comment by natedogs911** [comment]
+1

**Comment by solrac901** [comment]
+1

**Comment by jarcec** [comment]
+1

**Comment by natedogs911** [comment]
Hi @TaddWood The pr is merged, but the gitbot failed and the PR is still open.
Can you close this?

**Comment by TaddWood** [comment]
@natedogs911 no problem I will close the PR.  Thanks for the heads up.

**Pull request closed without merging**